src/core/detail/bitpacking.cc

```diff
@@ -104,6 +108,38 @@ static inline pair<const char*, uint8_t*> simd_variant2_pack(const char* ascii,
 // See https://github.com/lemire/fastvalidate-utf-8/
 // The function returns true (1) if all chars passed in src are
 // 7-bit values (0x00..0x7F). Otherwise, it returns false (0).
+#ifdef __s390x__
+bool validate_ascii_fast(const char* src, size_t len) {
+  size_t i = 0;
+
+  // Initialize a vector in which all the elements are set to zero.
+  vector signed char has_error = vec_splat_s8(0);
+  if (len >= 16) {
+    for (; i <= len - 16; i += 16) {
+      // Load 16 bytes from buffer into a vector.
+      vector signed char current_bytes = vec_load_len((signed char*)(src + i), 16);
+      // Perform a bitwise OR operation between the current and the previously loaded contents.
+      has_error = vec_orc(has_error, current_bytes);
+    }
+  }
+
+  int error_mask = 0;
+  for (int i = 0; i < 16; i++) {
+    if (has_error[i]) {
```

Comment:
hey, have you tested this function? 
according to the docs  https://doc.rust-lang.org/beta/core/arch/x86_64/fn._mm_movemask_epi8.html

_mm_movemask_epi8 returns the mask of msb bits (0x80). you do something else, IMHO.
Also, if you use scalar code here, there is no point in duplicating exactly the same logic really.
```
for (int i = 0; i < 16; i++) {
    if (has_error[i] & 0x80) 
      return false;
}
```

Reply:
Thanks for the comment. I fixed the logic according to your suggestion.
I've not found an implementation of the function '_mm_movemask_epi8 ' for s390x.